### PR TITLE
Provide interactive examples

### PR DESCRIPTION
**Pull request by dhodovsk**

### CODE REVIEW

File: conu/backend/docker/container.py [hunk 1]
```diff
@@ -370,3 +370,14 @@ def exit_code(self):
         :return: int
         """
         return self.get_metadata()["State"]["ExitCode"]
+
+    def active_write(self, message):
```
Comment (TomasTomecek) [inline conu/backend/docker/container.py]: Can we rename it to `write_to_stdin`? For me, the name is really confusing and I wouldn't know what I should expect.

File: conu/backend/docker/container.py [hunk 2]
```diff
@@ -370,3 +370,14 @@ def exit_code(self):
...
+        """
+        Send a message to running container
+
+        :param message: input to be written on container terminal
+        """
+        if not self.is_running():
+            raise ConuException("Container must be running")
+        self.popen_instance.stdin.write(message)
```
Comment (TomasTomecek) [inline conu/backend/docker/container.py]: please check that `self.popen_instance` is the object you wanted it to be; something like:

```python
if not self.popen_instance:
  raise ConuException("This container doesn't seem to be created using method `run_via_binary_in_foreground`.")
```

Comment (TomasTomecek) [inline conu/backend/docker/container.py]: And we should also probably catch exceptions from the write and reraise using our exception.

File: docs/source/examples/check_env.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import subprocess
+
+from conu import DockerBackend, DockerRunBuilder
+
+image = DockerBackend.ImageClass('fedora')
+cmd = DockerRunBuilder(additional_opts=['-i', '-e', 'KEY=space'])
+
+cont = image.run_via_binary_in_foreground(cmd, popen_params={"stdin": subprocess.PIPE})
```
Comment (TomasTomecek) [inline docs/source/examples/check_env.py]: I'm not sure if we have the stdin trick in our docs. It should definitely be mentioned in docstring of `run_via_binary_in_foreground`.

File: docs/source/examples/client-server-request.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import subprocess
+import logging
+from conu import DockerBackend, DockerRunBuilder
+
+environment = ["-e", "POSTGRESQL_USER=user",
+               "-e", "POSTGRESQL_PASSWORD=pass",
+               "-e", "POSTGRESQL_DATABASE=db"]
+
+backend = DockerBackend(logging_level=logging.DEBUG)
+image = backend.ImageClass('centos/postgresql-96-centos7')
```
Comment (TomasTomecek) [inline docs/source/examples/client-server-request.py]: I hope their image will never break :D It's definitely good to have such a brave test.

File: conu/backend/docker/container.py [hunk 5]
```diff
@@ -370,3 +370,14 @@ def exit_code(self):
         :return: int
         """
         return self.get_metadata()["State"]["ExitCode"]
+
+    def active_write(self, message):
+        """
+        Send a message to running container
+
+        :param message: input to be written on container terminal
```
[the remark this answers is not in the document]
Comment (TomasTomecek) [reply conu/backend/docker/container.py]: The rationale here is that you don't need allocate pty inside container to get interactivity.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/05UN](https://da.gd/05UN)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/FYjq](https://da.gd/FYjq)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/RKPvC](https://da.gd/RKPvC)__
* Comment: __See details for logs__

File: conu/backend/docker/container.py [hunk 6]
```diff
@@ -370,3 +371,36 @@ def exit_code(self):
...
+            raise ConuException(
+                "Container must be running")
+        if not self.popen_instance:
+            raise ConuException(
+                "This container doesn't seem to be created using method `run_via_binary_in_foreground`.")
+        if not self.popen_instance.stdin:
+            raise ConuException(
+                "Container should be run with stdin redirection.")
+
+        if not isinstance(message, bytes):
```
Comment (TomasTomecek) [inline conu/backend/docker/container.py]: +1

**Comment by TomasTomecek** [comment]
Thank you for these awesome changes. Tests are passing for me locally, merging.

**Pull request merged**